centreon/src/Core/HostGroup/Application/UseCase/UpdateHostGroup/UpdateHostGroup.php

```diff
@@ -0,0 +1,246 @@
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Application\Common\UseCase\ConflictResponse;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\Application\Common\UseCase\NoContentResponse;
+use Core\Application\Common\UseCase\NotFoundResponse;
+use Core\Application\Common\UseCase\PresenterInterface;
+use Core\Domain\Common\GeoCoords;
+use Core\Domain\Exception\InvalidGeoCoordException;
+use Core\HostGroup\Application\Exceptions\HostGroupException;
+use Core\HostGroup\Application\Repository\ReadHostGroupRepositoryInterface;
+use Core\HostGroup\Application\Repository\WriteHostGroupRepositoryInterface;
+use Core\HostGroup\Domain\Model\HostGroup;
+use Core\Infrastructure\Common\Api\DefaultPresenter;
+use Core\Security\AccessGroup\Application\Repository\ReadAccessGroupRepositoryInterface;
+use Core\ViewImg\Application\Repository\ReadViewImgRepositoryInterface;
+
+final class UpdateHostGroup
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private readonly ReadHostGroupRepositoryInterface $readHostGroupRepository,
+        private readonly WriteHostGroupRepositoryInterface $writeHostGroupRepository,
+        private readonly ReadAccessGroupRepositoryInterface $readAccessGroupRepository,
+        private readonly ReadViewImgRepositoryInterface $readViewImgRepository,
+        private readonly ContactInterface $contact
+    ) {
+    }
+
+    /**
+     * @param int $hostGroupId
+     * @param UpdateHostGroupRequest $request
+     * @param DefaultPresenter $presenter
+     */
+    public function __invoke(
+        int $hostGroupId,
+        UpdateHostGroupRequest $request,
+        PresenterInterface $presenter
+    ): void {
+        try {
+            if ($this->contact->isAdmin()) {
+                $response = $this->updateHostGroupAsAdmin($hostGroupId, $request);
+            } elseif ($this->contactCanPerformWriteOperations()) {
+                $response = $this->updateHostGroupAsContact($hostGroupId, $request);
+            } else {
+                $response = new ForbiddenResponse(HostGroupException::accessNotAllowedForWriting());
+            }
+
+            if ($response instanceof NoContentResponse) {
+                $presenter->present($response);
+                $this->info('Update host group', ['request' => $request]);
+            } elseif ($response instanceof NotFoundResponse) {
+                $presenter->setResponseStatus($response);
+                $this->warning('Host group (%s) not found', ['id' => $hostGroupId]);
+            } else {
+                $presenter->setResponseStatus($response);
+                $this->error(
+                    "User doesn't have sufficient rights to update host groups",
+                    ['user_id' => $this->contact->getId()]
+                );
+            }
```

Comment:
```suggestion
            } elseif ($response instanceof NotFoundResponse) {
                $this->warning('Host group (%s) not found', ['id' => $hostGroupId]);
            } else {
                $this->error(
                    "User doesn't have sufficient rights to update host groups",
                    ['user_id' => $this->contact->getId()]
                );
            }
           $presenter->setResponseStatus($response);
```

Reply:
To me : not worth. It does not improve readibility.